phoenixctms/ctsms
# EcrfDataEntryTest testsuite

this is the commit for our 2nd End-To-End testsuite based on chrome webdriver/selenium server.
it is to be executed automatically by the GitHub java CI (see "Actions" https://github.com/phoenixctms/ctsms/actions).

the use case covered is the "eCRF Date Entry and Export", as raised by AGES Inspektion KPAM 398.

Test outline:
- create phoenix instance, preload it with generated demo data
- setup records required for the test:
  - department
  - user
  - trial
- open the phoenix web UI in a chrome browser instance to programatically
  - log in
  - import sample eCRFs from a predefined test setup: ecrf.xls
    - cover all support field types
    - series sections
    - form scripting, calculated values
    - a new spreadsheet "validation" allows the specify what values to enter in the UI, and expect in the exported data
  - create a subject and add it to the trial's subject list
  - fill in each eCRF, with the data provided in the "validation" spreadsheet
  - run the eCRF data export job
  - download the exported data in SQLite format and verify the exported data:
    - compare each exported value with the reference data provided in the "validation" spreadsheet
    - check if all expected values are present
    - check if no unexpected values listed
- generate a test report and send results via email 
- upload the test reports to an FTP target forarchiving
  - the test reports for the given commit can also be linked in the login screen for download
 
An example of the generated test report PDF is attached below. It represent the documentation mandatory for **software validation** in regulated environments.

[Trial Tests.pdf](https://github.com/phoenixctms/ctsms/files/10814413/Trial.Tests.pdf)

Feel free to contact us if you need these test report documents as well.
